src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/SubmissionSummaryTable.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2022 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { StatArea } from "@/app/components/Table/common";
+import {
+  ColumnDef,
+  Table,
+  TableCellWrapper,
+  css,
+  useTheme,
+} from "@icgc-argo/uikit";
+import { capitalize } from "lodash";
+import { createRef } from "react";
+import { ClinicalEntity, ClinicalEntityType } from "../types";
+import { FILE_STATE_COLORS, RecordState } from "./FilesNavigator/StatsArea";
+
+type SubmissionSummaryStatus = {
+  submissionSummaryStatus: string;
+};
+
+// This type allows us to have a subset of the GqlClinicalEntity properties as columns, but not any other columns.
+//   A SubmissionSummaryStatus column is also allowed.
+type SubmissionSummaryColumns = Partial<
+  SubmissionSummaryStatus & {
+    [k in ClinicalEntityType]: string;
```

Comment:
Optionally you could write this as `Partial<SubmissionSummaryStatus & Record<ClinicalEntityType, string>>`

This is a personal preference thing but I find `Record<KeyType, ValueType>` easier to read than the `{ [k in KeyValues]: ValueType }` pattern.